frontend/src/scenes/funnels/funnelLogic.ts

```diff
@@ -1069,10 +1093,32 @@ export const funnelLogic = kea<funnelLogicType>({
             actions.loadEventWithPropertyCorrelations(eventName)
         },
 
-        excludeProperty: () => {
-            actions.setPropertyNames(
-                values.propertyNames.filter((property) => !values.excludedPropertyNames.includes(property))
-            )
+        excludePropertyFromProject: ({ propertyName }) => {
+            const oldExcludedPropertyNames = values.excludedPropertyNames
+            const oldCurrentTeam = teamLogic.values.currentTeam
+
+            if (oldCurrentTeam === null || oldExcludedPropertyNames === null) {
+                return
+            }
+
+            const oldCorrelationConfig = oldCurrentTeam.correlation_config
+
+            const excludedPropertyNames = [...Array.from(new Set(oldExcludedPropertyNames.concat([propertyName])))]
+
+            const correlationConfig = {
+                ...oldCorrelationConfig,
+                excluded_person_property_names: excludedPropertyNames,
+            }
+
+            teamLogic.actions.updateCurrentTeam({
+                correlation_config: correlationConfig,
+            })
+        },
+
+        [teamLogic.actionTypes.updateCurrentTeamSuccess]: async () => {
+            // When we have updated the current team, we'll have retrieved the
+            // excluded property names
+            actions.loadPropertyCorrelations()
```

Comment:
This is dicey: `updateCurrentTeamSuccess` is called whenever the team is updated. And we don't want to launch this action every time that happens, as it would lead to random `toast`s with errors on property correlations.

Hmm, is the problem right now that when you `loadPropertyCorrelations`, the new values of exclude aren't here yet?

Reply:
Yes, I could do an optimistic update and maintain some more state locally. What do you think?

Reply:
Alternatively, I'm thinking we don't need to immediately load correlations again on Exclude? Just remove from list?

Reply:
> some more state locally

As in, inside funnelLogic?

Feels more trouble than its worth, to be honest. If we update the selector in `propertyCorrelationValues` to use `excludedPropertyNames` as well, and filter on that, think we can achieve not loading again + ensuring there's no extra state? Since once the team is updated, the selector would update, which would then clean the excluded values.